tst/EnergyPlus/unit/PurchasedAirManager.unit.cc

```diff
@@ -1461,22 +1516,34 @@ TEST_F(ZoneIdealLoadsTest, IdealLoads_Fix_SA_HumRat_Test)
 
     CalcPurchAirLoads(*state, 1, SysOutputProvided, MoistOutputProvided, 1);
 
-    EXPECT_DOUBLE_EQ(state->dataPurchasedAirMgr->PurchAir(1).MinCoolSuppAirHumRat, 0.009);
+    EXPECT_DOUBLE_EQ(PurchAir.MinCoolSuppAirHumRat, 0.009);
 
-    EXPECT_DOUBLE_EQ(state->dataPurchasedAirMgr->PurchAir(1).SupplyTemp, 20.228931255157292);
+    EXPECT_DOUBLE_EQ(PurchAir.SupplyTemp, 20.228931255157292);
     // Without the current fix, this SupplyHumRat value would be 0.009, which is incorrect:
-    EXPECT_DOUBLE_EQ(state->dataPurchasedAirMgr->PurchAir(1).SupplyHumRat, 0.01);
+    EXPECT_DOUBLE_EQ(PurchAir.SupplyHumRat, 0.01);
 
-    EXPECT_DOUBLE_EQ(state->dataPurchasedAirMgr->PurchAir(1).MixedAirTemp, 30.0);
-    EXPECT_DOUBLE_EQ(state->dataPurchasedAirMgr->PurchAir(1).MixedAirHumRat, 0.012);
+    EXPECT_DOUBLE_EQ(PurchAir.MixedAirTemp, 30.0);
+    EXPECT_DOUBLE_EQ(PurchAir.MixedAirHumRat, 0.012);
 
-    EXPECT_DOUBLE_EQ(state->dataPurchasedAirMgr->PurchAir(1).SenCoilLoad, -1003.6327856486452);
-    EXPECT_DOUBLE_EQ(state->dataPurchasedAirMgr->PurchAir(1).LatCoilLoad, 5574.8612856486452);
+    EXPECT_DOUBLE_EQ(PurchAir.SenCoilLoad, -1003.6327856486452);
+    EXPECT_DOUBLE_EQ(PurchAir.LatCoilLoad, 5574.8612856486452);
 
-    EXPECT_DOUBLE_EQ(state->dataPurchasedAirMgr->PurchAir(1).SenOutputToZone, -1000.0000000000002);
-    EXPECT_DOUBLE_EQ(state->dataPurchasedAirMgr->PurchAir(1).LatOutputToZone, 5571.2285000000002);
+    EXPECT_DOUBLE_EQ(PurchAir.SenOutputToZone, -1000.0000000000002);
+    EXPECT_DOUBLE_EQ(PurchAir.LatOutputToZone, 5571.2285000000002);
 
     EXPECT_DOUBLE_EQ(state->dataLoopNodes->Node(1).Enthalpy, 45712.285000000003);
     EXPECT_DOUBLE_EQ(state->dataLoopNodes->Node(1).HumRat, 0.01);
     EXPECT_DOUBLE_EQ(state->dataLoopNodes->Node(1).Temp, 20.228931255157292);
+
+    ReportPurchasedAir(*state, 1);
+    EXPECT_EQ(PurchAir.heatFuelEffSched->getCurrentVal(), 2.0);
+    EXPECT_EQ(PurchAir.coolFuelEffSched->getCurrentVal(), 3.0);
+    EXPECT_DOUBLE_EQ(PurchAir.TotCoolRate, 1003.6327856486452);
+    EXPECT_DOUBLE_EQ(PurchAir.TotCoolFuelRate, 1003.6327856486452 / 3.0);
+    EXPECT_DOUBLE_EQ(PurchAir.TotHeatRate, 5574.8612856486452);
+    EXPECT_DOUBLE_EQ(PurchAir.TotHeatFuelRate, 5574.8612856486452 / 2.0);
+    EXPECT_DOUBLE_EQ(PurchAir.ZoneTotCoolRate, 1000.0000000000002);
+    EXPECT_DOUBLE_EQ(PurchAir.ZoneTotCoolFuelRate, 1000.0000000000002 / 3.0);
+    EXPECT_DOUBLE_EQ(PurchAir.ZoneTotHeatRate, 5571.2285000000002);
+    EXPECT_DOUBLE_EQ(PurchAir.ZoneTotHeatFuelRate, 5571.2285000000002 / 2.0);
```

Comment:
These checks demonstrate the fuel efficiency values after the override, as well as the values of the new report variable that has been added. The fuel energy consumption rates are obtained by dividing the loads by the fuel efficiency values.